### PR TITLE
Add iam_role_facts module

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
New iam_role_facts module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
iam_role_facts

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 90b867b7a5) last updated 2017/11/08 12:02:32 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.14 (default, Nov  2 2017, 18:42:05) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/iam_role_facts.py:11]: Is `certified` correct?

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            - Prefix of role I(path) to restrict IAM role search for
+            - Mutually exclusive with C(name)
+        required: false
+        default: None
+extends_documentation_fragment:
+    - aws
+'''
+
+EXAMPLES = '''
+# find all existing IAM roles
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/iam_role_facts.py:44]: Please use `-name: find all existing IAM roles` (and remove `-` from next line), same for other examples

**Review by gundalow — COMMENTED** [review]
Tests would be good :)

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py (hunk 1 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/iam_role_facts.py:11]: Umm, absolutely not. I wonder what I copied *that* from.

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py (hunk 2 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/iam_role_facts.py:44]: sure. I did have similar thoughts while editing whatever I copied it from :)

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py (hunk 1 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/iam_role_facts.py:11]: I hope Ansible know that `efs_facts` is a certified support module ;)

**Comment by willthames** [comment]
Nominal test suite added, there's really not much to test.

**Comment by willthames** [comment]
@mattclay @s-hertel this needs a few IAM read permissions (see the included iam-policy.json)

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                return []
+            else:
+                module.fail_json_aws(e, msg="Couldn't get IAM role %s" % name)
+        except botocore.exceptions.BotoCoreError as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM role %s" % name)
+    else:
+        params = dict()
+        if path_prefix:
+            params['PathPrefix'] = path_prefix
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/iam_role_facts.py]: http://docs.aws.amazon.com/IAM/latest/APIReference/API_ListRoles.html

I think we should enforce this so we can give a nice error for people that don't submit proper paths, or auto-fill it so the path is valid. Paths like `foobar/` and `/foobar` will break, the path *must* start and end with `/`. Up to you, but I'd just prepend and append `/` if there isn't one there already.

**Review by ryansb — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/amazon/iam_role_facts.py (hunk 3 above)
Comment (ryansb) [reply lib/ansible/modules/cloud/amazon/iam_role_facts.py]: Ping @willthames - other than this validation fix this is good to go.

Comment (willthames) [reply lib/ansible/modules/cloud/amazon/iam_role_facts.py]: @ryansb - thanks for that (and the ping) - I've updated as suggested.

**Review by ryansb — APPROVED** [review]

**Pull request merged**